src/sagemaker/estimator.py

```diff
@@ -158,9 +154,14 @@ def fit(self, inputs, wait=True, logs=True, job_name=None):
             base_name = self.base_job_name or base_name_from_image(self.train_image())
             self._current_job_name = name_from_base(base_name)
 
-        # if output_path was specified we use it otherwise initialize here
+        # if output_path was specified we use it otherwise initialize here.
+        # For Local Mode with no_internet=True we don't need an explicit output_path
         if self.output_path is None:
-            self.output_path = 's3://{}/'.format(self.sagemaker_session.default_bucket())
+            no_internet = get_config_value('local.no_internet', self.sagemaker_session.config)
```

Comment:
This property should be local.local_code and it should default to true. Customers can still call out to the internet if 'no_internet' is true.

Reply:
much better, I was not a fan of no_internet.